[feat] add documents for QA (#47)
dwarvesf/playbook: engineering/Testing/defect-template.md

### Defect Report template

### Summary

> A brief summary of the bug, mostly within 60 words or below. Make sure your summary is reflecting on what the problem is and where it is.

Reviewer: ```suggestion
A summary of the bug should be no more than 60 words. Make sure the summary reflects the problem and where it is.
```

**What - Where - When (Action)**

### Description:

A detailed description of the bug.

Use the following fields for the description field:

**Reproduce steps:**

> Clearly, mention the steps to reproduce the bug.
> Example:

Reviewer: ```suggestion
Clearly mention the steps to reproduce the bug. For example:
```

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'

> 4. See error

Reviewer 2: For this one, I would prefer the text like `Observed result` - it would be better

Author: Updated

**Expected result:** How the application should behave on the above-mentioned steps.

**Actual result:** What is the actual result of running the above steps i.e. the bug behavior.

> **Screenshots**
> 
> If applicable, add screenshots to help explain your problem.

Reviewer: ```suggestion
**Screenshots**

If applicable, add screenshots or screen recordings to explain your problem.
```

**Desktop (please complete the following information):**

- OS: [e.g. iOS]
- Browser [e.g. chrome, safari]
- Version [e.g. 22]

**Smartphone (please complete the following information):**

- Device: [e.g. iPhone6]
- OS: [e.g. iOS8.1]
- Browser [e.g. stock browser, safari]
- Version [e.g. 22]

### Logs / Stack trace

```
Insert your log/stack trace here
```

Besides that, we also need to add more about Priority and Severity:

**Priority:** When should a bug be fixed? Priority is generally set from P1 to P5. P1 as “fix the bug with the highest priority” and P5 as ” Fix when time permits”.

**Severity:** This describes the impact of the bug.

**Types of Severity:**

Blocker: No further testing work can be done.
Critical: Application crash, Loss of data.
Major: Major loss of function.
Minor: Minor loss of function.
Trivial: Some UI enhancements.
Enhancement: Request for a new feature or some enhancement in the existing one.

===============================================================================

### Some Bonus Tips To Write A Good Bug Report

Given below are some additional tips on how to write a good Bug report:

> #1) Report the problem immediately

Reviewer: ```suggestion
#### Report the problem immediately
```

> If you find any bugs while testing, then you do not need to wait to write a detailed bug report later.

Reviewer 3: I think leave the title only. This one "Report the problem immediately" is meaningful enough

#2) Reproduce the bug three times before writing a Bug report detail
Your bug should be reproducible. Make sure that your steps are robust enough to reproduce the bug without any ambiguity. If your bug is not reproducible every time, then you can still file a bug mentioning the periodic nature of the bug.

#3) Test the same bug occurrence on other similar modules

> Sometimes the developer uses the same code for different similar modules. So there is a higher chance for the bug in one module to occur in other similar modules as well. You can even try to find the more severe version of the bug you found.

Author: I thinks your comment is related to create the test scenario.

#4) Write a good bug summary

> Bug summary will help the developers to quickly analyze the bug’s nature. A poor quality report will unnecessarily increase development and testing time. Communicate well with your bug report summary. Keep in mind that the bug summary can be used as a reference to search for the bug in the bug inventory.

Reviewer 3: Communicate well with your bug report summary. --> What does this mean?

Author: If the bug summary is good, dev team will easily to fix it. If not, they should be scanned to Description or Attachment to understand the problem.
